SYCL/Matrix/joint_matrix_all_sizes_impl.hpp

```diff
@@ -0,0 +1,195 @@
+    for (int j = 0; j < MATRIX_K; j++) {
+      if (std::is_same_v<Ta, bfloat16> && std::is_same_v<Tc, float>)
+        A[i][j] = bfloat16(1.0f * (i + j));
+      if (std::is_same_v<Ta, int8_t> && std::is_same_v<Tc, int32_t>)
+        A[i][j] = i + j;
+    }
+  }
+  for (int i = 0; i < MATRIX_K; i++) {
+    for (int j = 0; j < MATRIX_N; j++) {
+      if (std::is_same_v<Ta, bfloat16> && std::is_same_v<Tc, float>)
+        B[i][j] = bfloat16(2.0f * i + 3.0f * j);
+      if (std::is_same_v<Ta, int8_t> && std::is_same_v<Tc, int32_t>)
+        B[i][j] = i + 2 * j;
+    }
+  }
+  for (int i = 0; i < MATRIX_M; i++) {
+    for (int j = 0; j < MATRIX_N; j++) {
+      C[i][j] = 1;
+      D[i][j] = 1;
+    }
+  }
+
+  big_matrix<Tc, MATRIX_M, MATRIX_N> MC((Tc *)&C);
+  big_matrix<Tc, MATRIX_M, MATRIX_N> MD((Tc *)&D);
+  big_matrix<Ta, MATRIX_M, MATRIX_K> MA((Ta *)&A);
+  matrix_vnni<Ta>(MATRIX_K, MATRIX_N, (Ta *)&B, (Ta *)&Bvnni, vnni_factor);
+  big_matrix<Ta, MATRIX_K / vnni_factor, MATRIX_N * vnni_factor> MBvnni(
+      (Ta *)&Bvnni);
+
+  matrix_multiply<Tc, Ta, MATRIX_M, MATRIX_N, MATRIX_K, vnni_factor, tM, tN,
+                  tK>(MC, MA, MBvnni);
+  matrix_multiply_ref((Ta *)A, (Ta *)B, (Tc *)D, MATRIX_M, MATRIX_N, MATRIX_K);
+
+  bool res = true;
+  for (int i = 0; i < MATRIX_M; i++) {
+    for (int j = 0; j < MATRIX_N; j++) {
+      if constexpr (std::is_same_v<Ta, bfloat16> && std::is_same_v<Tc, float>) {
+        if ((fabs(C[i][j]) - fabs(D[i][j])) > BF16_EPSILON) {
+          std::cout << (res ? "passed" : "failed bfloat  ") << C[i][j]
+                    << " D is " << D[i][j] << std::endl;
+          res = false;
+        }
+      } else if (std::is_same_v<Ta, int8_t> && std::is_same_v<Tc, int32_t>) {
+        if (C[i][j] != D[i][j]) {
+          std::cout << (res ? "passed" : "failed") << C[i][j] << " D is "
+                    << D[i][j] << std::endl;
+
+          res = false;
+        }
+      }
+    }
+  }
+  std::cout << (res ? "passed" : "failed") << std::endl;
+  return !res;
+}
+
+int main() {
+  // init_and_multiply<bfloat16, float, 2, 1, 16, 16>();
+  // init_and_multiply<bfloat16, float, 2, 2, 16, 16>();
+  // init_and_multiply<bfloat16, float, 2, 3, 16, 16>();
+  init_and_multiply<bfloat16, float, 2, 4, 16, 16>();
+  // init_and_multiply<bfloat16, float, 2, 5, 16, 16>();
+  // init_and_multiply<bfloat16, float, 2, 6, 16, 16>();
+  // init_and_multiply<bfloat16, float, 2, 7, 16, 16>();
+  // init_and_multiply<bfloat16, float, 2, 8, 16, 16>();
+
+  // init_and_multiply<int8_t, int32_t, 4, 1, 16, 32>();
+  // init_and_multiply<int8_t, int32_t, 4, 2, 16, 32>();
+  // init_and_multiply<int8_t, int32_t, 4, 3, 16, 32>();
+  // init_and_multiply<int8_t, int32_t, 4, 4, 16, 32>();
+  // init_and_multiply<int8_t, int32_t, 4, 5, 16, 32>();
+  // init_and_multiply<int8_t, int32_t, 4, 6, 16, 32>();
+  // init_and_multiply<int8_t, int32_t, 4, 7, 16, 32>();
+  // init_and_multiply<int8_t, int32_t, 4, 8, 16, 32>();
```

Comment:
why do we have many lines commented out? What we need/wait to enable them?

Reply:
I uncommented them. All these calls with different sizes need support to remove the xfail